apollo-router/src/plugins/connectors/tests/query_plan.rs

```diff
@@ -0,0 +1,253 @@
+use wiremock::Mock;
+use wiremock::MockServer;
+use wiremock::ResponseTemplate;
+use wiremock::matchers::method;
+use wiremock::matchers::path;
+
+use super::req_asserts::Matcher;
+use super::req_asserts::Plan;
+
+#[tokio::test]
+async fn basic_batch() {
+    let mock_server = MockServer::start().await;
+    Mock::given(method("GET"))
+        .and(path("/users"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(serde_json::json!([
+        { "id": 3 },
+        { "id": 1 },
+        { "id": 2 }])))
+        .mount(&mock_server)
+        .await;
+    Mock::given(method("POST"))
+        .and(path("/users-batch"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(serde_json::json!([
+        {
+          "id": 1,
+          "name": "Leanne Graham",
+          "username": "Bret"
+        },
+        {
+          "id": 2,
+          "name": "Ervin Howell",
+          "username": "Antonette"
+        },
+        {
+          "id": 3,
+          "name": "Clementine Bauch",
+          "username": "Samantha"
+        }])))
+        .mount(&mock_server)
+        .await;
+
+    let response = super::execute(
+        include_str!("../testdata/batch.graphql"),
+        &mock_server.uri(),
+        "query { users { id name username } }",
+        Default::default(),
+        Some(serde_json_bytes::json!({
+            "plugins": {
+              "experimental.expose_query_plan": true
+            }
+        })),
+        |req| {
+            req.router_request
+                .headers_mut()
+                .append("Apollo-Expose-Query-Plan", "true".parse().unwrap());
+        },
+    )
+    .await;
+
+    insta::assert_json_snapshot!(response, @r#"
```

Comment:
Shouldn't this be saved as a file somewhere?

Reply:
the `@` makes it an inline snapshot

Reply:
https://insta.rs/docs/snapshot-types/#inline-snapshots